cmd/libs/go2idl/client-gen/test_apis/testgroup/v1/types.go

```diff
@@ -32,6 +33,8 @@ type TestType struct {
 	Status TestTypeStatus `json:"status,omitempty"`
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

Comment:
Every (api) type gets a DeepCopy func, but we have runtime.Objects and selectors which are used in a polymorphic way. I.e. you get a runtime.Object without knowing which type it really is. For top-level API objects we create an additional type-independent `DeepCopyObject() runtime.Object` method. My original proposal was adding these automatically for those types embedding `TypeMeta` (compare https://github.com/kubernetes/gengo/pull/44). It was decided by @thockin and @smarterclayton that this was too much magic. So we have these annotations now which trigger the creation of `DeepCopyObject() runtime.Object` explicitly.